index.js

```diff
@@ -0,0 +1,418 @@
+	if (req.session.user) {
+		con.query("select * from booking;", (err, bookingData) => {
+			con.query("select * from events;", (err, eventsData) => {
+				res.render("dashboard_facility_card", {
+					res: req.session.user,
+					bookingData,
+					eventsData
+				});
+			});
+		});
+	} else {
+		res.redirect("/login");
+	}
+});
+app.get("/dashboard_facility_table", (req, res) => {
+	if (req.session.user) {
+		con.query("select * from booking;", (err, bookingData) => {
+			con.query("select * from events;", (err, eventsData) => {
+				res.render("dashboard_facility_table", {
+					res: req.session.user,
+					bookingData,
+					eventsData
+				});
+			});
+		});
+	} else {
+		res.redirect("/login");
+	}
+});
+app.get("/request", (req, res) => {
+	if (req.session.user) {
+		res.render("request");
+	} else {
+		res.redirect("/login");
+	}
+});
+app.post("/requestUpdate", (req, res) => {
+	if (req.session.user) {
+		con.query("select * from booking where id=?;", req.body.bookingID, (err, bookingData) => {
+			con.query("select * from events where id=?;", req.body.eventID, (err, eventsData) => {
+				eventsData[0].dateWanted =
+					eventsData[0].dateWanted.getFullYear() +
+					"/" +
+					eventsData[0].dateWanted.getMonth() +
+					"/" +
+					eventsData[0].dateWanted.getDate();
+				res.render("requestUpdate", {
+					res: req.session.user,
+					bookingData,
+					eventsData
+				});
+			});
+		});
+	} else {
+		res.redirect("/login");
+	}
+});
+app.get("/logout", (req, res) => {
+	res.clearCookie("session", {
+		path: "/"
+	});
+	res.redirect("/login");
+});
+app.post("/onLogin", (req, res) => {
+	con.query(
+		"select * from users where username= ?",
+		req.body.username,
+		(err, rows) => {
+			if (!err) {
+				if (rows.length > 0) {
+					if (
+						req.body.password ===
+						AES.decrypt(rows[0].password, `${process.env.AES_key}`)
+					) {
+						req.session.user = rows[0];
+						res.redirect("/dashboard");
+					} else {
+						console.log("Invalid password");
+						res.status(401).render("error", {
+							errorMessage: "Authentication error!"
+						});
+					}
+				} else {
+					console.log("Username not found!");
+					res.status(401).render("error", {
+						errorMessage: "Authentication error!"
+					});
+				}
+			} else {
+				console.log("Username not found!");
+				res.status(401).render("error", {
+					errorMessage: "Authentication error!"
+				});
+			}
+		}
+	);
+});
+app.post("/checkAvailability", (req, res) => {
+	if (req.session.user) {
+		con.query(
+			"SELECT slotHall FROM booking WHERE eventID IN (SELECT id FROM events WHERE dateWanted = ?)",
+			req.body.dateWanted,
+			(err, result) => {
+				if (!err) {
+					res.send(result);
+				} else {
+					console.log(err);
+					res.status(500).render("error", {
+						errorMessage: "Availability checking failed!"
+					});
+				}
+			}
+		);
+	} else {
+		res.redirect("/login");
+	}
+});
+app.post("/makeRequest", (req, res) => {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Arrow function has a complexity of 5. (complexity)](https://app.codacy.com/app/AravindVNair99/Hall-Management-System/pullRequest?prid=4592231)